qemu/tests/qemu_guest_agent.py

```diff
@@ -818,6 +818,37 @@ def _action_after_fsfreeze(self, *args):
             raise error.TestFail("FS freeze failed, guest still can"
                                  " write file")
 
+    def gagent_check_reset_reboot_shutdown(self, test, params, env):
+
+        """
+        Execute "system_reset" and send "shutdown,reboot"
+        command to guest agent after FS freezed
+        :param test: kvm test object
+        :param params: Dictionary with the test parameters
+        :param env: Dictionary with test environment.
+        """
+        vm = env.get_vm(params["main_vm"])
+        vm.verify_alive()
+        gagent = self.gagent
+        gagent.fsfreeze()
+
+        try:
+            if gagent.shutdown(gagent.SHUTDOWN_MODE_POWERDOWN):
+                test.fail("agent shutdown command shouldn't succeed for freeze FS")
+        except gagent.VAgentCmdError as detail:
```

Comment:
As you use if gagent.shutdown(gagent.SHUTDOWN_MODE_POWERDOWN), the VAgentCmdError will never be caught, please have a check.

Reply:
@lijinlijin Fixed this issue. please check it again. Thanks.